socs/agents/hwp_supervisor/agent.py

```diff
@@ -929,11 +1012,9 @@ def pid_to_freq(self, session, params):
             freq_tol_duration=params['freq_tol_duration'],
             direction=d
         )
-        success = self.control_state_machine.request_state(state)
-        if success:
-            return True, f"Set state to {state}"
-        else:
-            return False, "Failed to update state"
+        action = self.control_state_machine.request_new_action(state)
+        action.sleep_until_complete(session=session)
```

Comment:
Add an example of the `session.data` to the docstring.

(Do this everywhere `action.sleep_until_complete(session=session)` occurs.)

Reply:
Is it ok if I do something simple like:
```
>> session['data'] 
{'action': 
  {'action_id': 3,
  'completed': True,
  'cur_state': {'class': 'Done', 'msg': None, 'success': True},
  'state_history': List[ConrolState],
  'success': True}
}
```

Or do I need to go through and run each of the tasks and copy a realistic action for each task? It is a bit difficult to maintain correct session.data in per-operation docstrings, and there are already examples of what the encoded action looks like in the rst.

Reply:
Thanks for adding. I think simple examples are fine, as long as they contain enough information that someone developing a control program can know what to expect from the object.

Since the `session.data` object is sort of our API for passing data to clients for programmatic checks in control programs, these interfaces are important. It's difficult when writing a control program to use the `session.data` if it's not well documented, especially so if you don't have a running instance of the Agent to develop against.

On that note, changing what's kept in `session.data` should be carefully considered and then called out in future PRs, since those changes potentially break control programs that depend on them.